lib/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,623 @@
+                'Transition lifecycle action not yet implemented');
+            return ruleObj;
+        }
+        if ((!rule.Filter && rule.Filter !== '') &&
+        (!rule.Prefix && rule.Prefix !== '')) {
+            ruleObj.error = errors.MalformedXML.customizeDescription(
+                'Rule xml does not include valid Filter or Prefix');
+            return ruleObj;
+        }
+        if (rule.Filter && rule.Prefix) {
+            ruleObj.error = errors.MalformedXML.customizeDescription(
+                'Rule xml should not include both Filter and Prefix');
+            return ruleObj;
+        }
+        if (!rule.Status) {
+            ruleObj.error = errors.MalformedXML.customizeDescription(
+                'Rule xml does not include Status');
+            return ruleObj;
+        }
+        const subFilter = rule.Filter ? rule.Filter[0] : rule.Prefix;
+
+        const id = this._parseID(rule.ID);
+        const status = this._parseStatus(rule.Status[0]);
+        const filter = this._parseFilter(subFilter);
+        const actions = this._parseAction(rule);
+
+        const rulePropArray = [id, status, filter, actions];
+        rulePropArray.forEach(prop => {
+            if (prop.error) {
+                ruleObj.error = prop.error;
+            } else {
+                const propName = prop.propName;
+                // eslint-disable-next-line no-param-reassign
+                delete prop.propName;
+                ruleObj[propName] = prop[propName] || prop;
+            }
+        });
+        return ruleObj;
+    }
+
+    /**
+     * Check that filter component of rule is valid
+     * @param {object} filter - filter object from a rule object
+     * @return {object} - contains error if parsing failed, else contains
+     * parsed prefix and tag array
+     *
+     * Format of filterObj:
+     * filterObj = {
+     *      error: <error>,
+     *      propName: 'filter',
+     *      rulePrefix: <value>,
+     *      tags: [
+     *          {
+     *              key: <value>,
+     *              val: <value>
+     *          },
+     *          {
+     *              key: <value>,
+     *              value: <value>
+     *          }
+     *      ]
+     * }
+     */
+    _parseFilter(filter) {
+        const filterObj = {};
+        filterObj.propName = 'filter';
+        // if no Rule Prefix or Filter, rulePrefix is empty string
+        filterObj.rulePrefix = '';
+        if (Array.isArray(filter)) {
+            // if Prefix was included, not Filter, filter will be Prefix array
+            if (filter.length > 1) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'Rule includes more than one Prefix');
+                return filterObj;
+            }
+            filterObj.rulePrefix = filter[0];
+            return filterObj;
+        }
+        if (filter.And && (filter.Prefix || filter.Tag) ||
+        (filter.Prefix && filter.Tag)) {
```

Comment:
I'm guessing checking from left to right, but just to be safe, maybe add an outer parens around the first `&&` check